src/Microsoft.DocAsCode.Common/Loggers/HtmlLogListener.cs

```diff
@@ -0,0 +1,150 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Common
+{
+    using System;
+    using System.IO;
+    using System.Text.RegularExpressions;
+
+    using Newtonsoft.Json;
+
+    public sealed class HtmlLogListener : ILoggerListener
+    {
+        private readonly StreamWriter _writer;
+
+        private const LogLevel LogLevelThreshold = LogLevel.Diagnostic;
+
+        private static readonly Regex EscapeWithEncode = new Regex(@"&", RegexOptions.Compiled);
+        private static readonly Regex EscapeWithoutEncode = new Regex(@"&(?!#?\w+;)", RegexOptions.Compiled);
+
+#if !NetCore
+        public HtmlLogListener(string reportPath)
+        {
+            var dir = Path.GetDirectoryName(reportPath);
+            if (!string.IsNullOrEmpty(dir))
+            {
+                Directory.CreateDirectory(dir);
+            }
+            _writer = new StreamWriter(reportPath, true);
+            WriteCommonHeader();
+        }
+#endif
+
+        public HtmlLogListener(StreamWriter writer)
+        {
+            if (writer == null)
+            {
+                throw new ArgumentNullException(nameof(writer));
+            }
+            _writer = writer;
+            WriteCommonHeader();
+        }
+
+        public void WriteCommonHeader()
+        {
+            _writer.WriteLine("<html>");
+            _writer.WriteLine(@"<body lang=""en-us"">");
+            _writer.WriteLine(@"<h1 id=""Report"">Report</h1>");
+            _writer.WriteLine(@"<table border=""1"">");
+            _writer.WriteLine(@"<tr><td>Severity</td><td>Message</td><td>File</td><td>Line</td><td>Time</td></tr>");
+        }
+
+        public void WriteLine(ILogItem item)
+        {
+            if (item == null)
+            {
+                throw new ArgumentNullException(nameof(item));
+            }
+            var level = item.LogLevel;
+            var message = item.Message;
+            var phase = item.Phase;
+            var file = item.File;
+            var line = item.Line;
+            if (level < LogLevelThreshold) return;
+
+            var reportItem = new ReportItem
+            {
+                Severity = GetSeverity(level),
+                Message = message,
+                Source = phase,
+                File = file,
+                Line = line,
+                DateTime = DateTime.UtcNow
+            };
+
+            _writer.WriteLine($@"<tr><td>{reportItem.Severity}</td><td>{Escape(reportItem.Message)}</td><td>{reportItem.File}</td><td>{reportItem.Line}</td><td>{reportItem.DateTime}</td></tr>");
```

Comment:
Escape File. Line is not necessary to escape.

---
In reply to: [74915055](https://github.com/dotnet/docfx/pull/605#discussion_r74915055) [](ancestors = 74915055)